src/libraries/System.IO.FileSystem.AccessControl/tests/FileSystemAclExtensionsTests.cs

```diff
@@ -176,80 +179,107 @@ public void SetAccessControl_FileStream_FileSecurity_Success()
-            DirectoryInfo info = new DirectoryInfo(path);
-            DirectorySecurity security = new DirectorySecurity();
+            using var tempRootDir = new TempAclDirectory();
+            string dirPath = Path.Combine(tempRootDir.Path, Guid.NewGuid().ToString(), "ParentDoesNotExist");
+
+            var dirInfo = new DirectoryInfo(dirPath);
+            var security = new DirectorySecurity();
+            // Fails because the DirectorySecurity lacks any rights to create parent folder
+            Assert.Throws<UnauthorizedAccessException>(() =>  CreateDirectoryWithSecurity(dirInfo, security));
+        }
+
+        [Fact]
+        public void DirectoryInfo_Create_NotFound_FullControl()
+        {
+            using var tempRootDir = new TempAclDirectory();
+            string dirPath = Path.Combine(tempRootDir.Path, Guid.NewGuid().ToString(), "ParentDoesNotExist");
 
-            Assert.Throws<UnauthorizedAccessException>(() =>
+            var dirInfo = new DirectoryInfo(dirPath);
+            var security = GetDirectorySecurity(FileSystemRights.FullControl);
+            // Succeeds because it creates the missing parent folder
+            CreateDirectoryWithSecurity(dirInfo, security);
+        }
+
+        private void CreateDirectoryWithSecurity(DirectoryInfo info, DirectorySecurity security)
+        {
+            if (PlatformDetection.IsNetFramework)
             {
-                if (PlatformDetection.IsNetFramework)
-                {
-                    FileSystemAclExtensions.Create(info, security);
-                }
-                else
-                {
-                    info.Create(security);
-                }
-            });
+                FileSystemAclExtensions.Create(info, security);
+            }
+            else
+            {
+                info.Create(security);
+            }
         }
 
         [Fact]
         public void DirectoryInfo_Create_DefaultDirectorySecurity()
         {
-            DirectorySecurity security = new DirectorySecurity();
-            Verify_DirectoryInfo_Create(security);
+            var security = new DirectorySecurity();
+            Verify_DirectorySecurity_CreateDirectory(security);
         }
 
         [Theory]
-        [InlineData(FileSystemRights.ReadAndExecute, AccessControlType.Allow)]
-        [InlineData(FileSystemRights.ReadAndExecute, AccessControlType.Deny)]
-        [InlineData(FileSystemRights.WriteData,      AccessControlType.Allow)]
-        [InlineData(FileSystemRights.WriteData,      AccessControlType.Deny)]
-        [InlineData(FileSystemRights.FullControl,    AccessControlType.Allow)]
-        [InlineData(FileSystemRights.FullControl,    AccessControlType.Deny)]
-        public void DirectoryInfo_Create_DirectorySecurityWithSpecificAccessRule(
-            FileSystemRights rights,
-            AccessControlType controlType)
+        // Must have at least one Read, otherwise the TempAclDirectory will fail to delete that item on dispose
```

Comment:
Is this comment still true?

Reply:
Yes

Reply:
What does it mean to have at least one Read? It means that every `FileSystemRights` parameter passed to this test should contain the flag `Read`?

Reply:
Correct. Actually, it should specifically be ReadData.